### PR TITLE
Random mode modification

### PR DESCRIPTION
**Pull request by sepandhaghighi**
#### What does this implement/fix? Explain your changes.
- Random mode modified

### CODE REVIEW

File: samila/functions.py [hunk 1]
```diff
@@ -27,17 +27,22 @@ def random_equation_gen():
 
     :return: equation as str
     """
-    num_elements = random.randint(2, len(ELEMENTS_LIST) + 3)
```
Comment (sadrasabouri) [inline samila/functions.py:30]: Why did we do that earlier? :thinking:

File: samila/functions.py [hunk 2]
```diff
@@ -27,17 +27,22 @@ def random_equation_gen():
...
+        if random.randint(0, 1) == 1:
+            argument = random.choice(ELEMENTS_LIST).format(
+                random_coef, argument)
         result = result + \
             random.choice(ELEMENTS_LIST).format(random_coef, argument)
         if index < num_elements:
             result = result + random.choice(OPERATORS_LIST)
         index = index + 1
+    if random.randint(0, 1) == 1:
+        result = random.choice(ELEMENTS_LIST).format(random_coef, result)
```
Comment (sadrasabouri) [inline samila/functions.py:45]: This part is a bit odd for me. Please provide more documents for these changes here for the record.

**Review by sadrasabouri — CHANGES_REQUESTED** [review]
Thank you for the PR. I left minor comments.
[JFR: I think we can have another function like `random_equation_gen` which uses context free grammars as input and make run a random walk on it, by this function we can map a CFG to a family of Samila arts, too]

File: samila/functions.py (hunk 2 above)
Comment (sepandhaghighi) [reply samila/functions.py:45]: It's some kind of `fof` (function composition)

File: samila/functions.py (hunk 1 above)
Comment (sepandhaghighi) [reply samila/functions.py:30]: In the early versions of the `random_equation_gen` function, the length of `ELEMENTS_LIST` was about 2-3, with the addition of more options to this list we need to reduce the maximum number of elements (to balance complexity and beauty)

**Review by sadrasabouri — APPROVED** [review]

**Pull request merged**